[receiver/mongodbatlas] Safeguard polled alerts potentially overloading exporters (#14611)
open-telemetry/opentelemetry-collector-contrib: receiver/mongodbatlasreceiver/README.md

  - `secret` (required if using `listen` mode)
  - `endpoint` (required if using `listen` mode)
  - `poll_interval` (default `5m`, only relevant using `poll` mode)
  - `page_size` (default `100`)

>     - When in `poll` mode, this is the number of alerts that will be processed per request to the MongoDB Atlas API per project.

Reviewer: ```suggestion
    - When in `poll` mode, this is the number of alerts that will be processed per request to the MongoDB Atlas API.
```
This isn't really a variable in the context of projects. It's just going to always be the page size we request.

  - `max_pages` (default `10`)

>     - When in `poll` mode, this will limit how many pages of alerts the receiver will request to process the alerts.

Reviewer: ```suggestion
    - When in `poll` mode, this will limit how many pages of alerts the receiver will request for each project.
```

This one is actually really important to note that it is per project.

Author: good point 👍
